lib/compress/zstd_compress.c

```diff
@@ -1935,7 +1935,18 @@ ZSTD_reset_matchState(ZSTD_matchState_t* ms,
     if (ZSTD_rowMatchFinderUsed(cParams->strategy, useRowMatchFinder)) {
         /* Row match finder needs an additional table of hashes ("tags") */
         size_t const tagTableSize = hSize * sizeof(U16);
-        ms->tagTable = (U16 *) ZSTD_cwksp_reserve_aligned_init_once(ws, tagTableSize);
+        /* We want to generate a new salt in case we reset a Cctx, but we always want to use
+         * 0 when we reset a Cdict */
+        if(forWho == ZSTD_resetTarget_CCtx) {
+            ms->tagTable = (U16 *) ZSTD_cwksp_reserve_aligned_init_once(ws, tagTableSize);
+            ms->hashSalt = (U64) ZSTD_hashPtr(&ms->hashSalt, 32, sizeof(ms->hashSalt)) << 32 |
```

Comment:
I would rather go for a stronger mixing, making observation of any portion of bits essentially random (including lower bits).

I would recommend to base it on  `XXH3_rrmxmx()` :
https://github.com/Cyan4973/xxHash/blob/v0.8.1/xxhash.h#L3403

This mixing has been tested and proved strong enough to pass `practrand` tests,
making it essentially as strong as a PCG random number generator.